src/main/java/com/myaudiolib/web/controller/ArtistController.java

```diff
@@ -0,0 +1,119 @@
+package com.myaudiolib.web.controller;
+
+import com.myaudiolib.web.model.Artist;
+import com.myaudiolib.web.repository.ArtistRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.view.RedirectView;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.Optional;
+
+
+@Controller
+@RequestMapping("/artists")
+public class ArtistController {
+
+    @Autowired
+    private ArtistRepository artistRepository;
+
+    // Récupérer un artiste par son id
+    @GetMapping(value = "/{id}")
+    public String detailArtist(
+            @PathVariable Long id,
+            final ModelMap model) {
+        Optional<Artist> artistOptional = artistRepository.findById(id);
+
+        // gestion de l'erreur 404
+        if ((artistOptional == null) || (artistOptional.isEmpty())){
+            throw new EntityNotFoundException("L'artiste dont l'identifiant est " + id + "  n'a pas été trouvé.");
+        }
+        model.put("artist", artistOptional.get());
+        return "detailArtist";
+    }
+
+
+    // Récupérer un artiste par nom
+    @GetMapping(value = "",
+            params = {"name"})
+    public String searchArtistByName(
+            @RequestParam(value = "name") String name,
+            final ModelMap model){
+        List<Artist> artistList = artistRepository.findAllByNameContaining(name);
+        // gestion de l'erreur 404
+        if (artistList == null) {
+            throw new EntityNotFoundException("L'artiste " + name + " recherché n'a pas été trouvé.");
+        }
```

Comment:
Attention la liste peut être non null est vide. 
```suggestion
        if (artistList == null || artistList.isEmpty()) {
            throw new EntityNotFoundException("L'artiste " + name + " recherché n'a pas été trouvé.");
        }
```